libcontainer/rootfs_linux.go

```diff
@@ -723,7 +724,13 @@ func readonlyPath(path string) error {
 		}
 		return err
 	}
-	return unix.Mount(path, path, "", unix.MS_BIND|unix.MS_REMOUNT|unix.MS_RDONLY|unix.MS_REC, "")
+
+	statfs := syscall.Statfs_t{}
+	syscall.Statfs(path, &statfs)
+
+	flags := statfs.Flags | unix.MS_BIND | unix.MS_REMOUNT | unix.MS_RDONLY | unix.MS_REC
```

Comment:
We don't use `syscall` anymore, we use the extended-stdlib `golang.org/x/sys/unix` library. Also you need to handle the error. So this should look more like:

```
var statfs unix.Statfs_t
if err := unix.Statfs(path, &statfs); err != nil {
	return err
}

statfs.Flags |= unix.MS_BIND | unix.MS_REMOUNT | unix.MS_RDONLY | unix.MS_REC
return unix.Mount(path, path, "", uintptr(statfs.Flags), "")
```

Reply:
Also it looks like these changes weren't correctly `gofmt`'d. Run `gofmt -s -w .` to fix this up.